selectrum.el

```diff
@@ -414,7 +441,8 @@ Passed to various hook functions.")
   "Clean up Selectrum from the minibuffer, and self-destruct this hook."
   (remove-hook
    'post-command-hook #'selectrum--minibuffer-post-command-hook 'local)
-  (remove-hook 'minibuffer-exit-hook #'selectrum--minibuffer-exit-hook 'local))
+  (remove-hook 'minibuffer-exit-hook #'selectrum--minibuffer-exit-hook 'local)
+  (setq selectrum--count-overlay nil))
```

Comment:
We should also make sure to `delete-overlay` during cleanup, right?

Reply:
I don't know. Is setting it to `nil` not enough? I'd like to hear your explanation.

Reply:
That doesn't actually delete the overlay. It just forgets about the reference to the variable. The code you have right now relies on the minibuffer getting deleted (and hence the overlay being removed from it), but I would prefer to actually delete the overlay (which removes it from the buffer) explicitly.

You can see the distinction if you create an overlay in a regular file buffer. It won't go away unless you call `delete-overlay`.

Reply:
Done.